packages/core/upload/admin/src/components/UploadAssetDialog/UploadAssetDialog.js

```diff
@@ -1,85 +1,28 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import { ModalLayout, ModalHeader, ModalFooter, ModalBody } from '@strapi/parts/ModalLayout';
-import { ButtonText } from '@strapi/parts/Text';
-import { Button } from '@strapi/parts/Button';
-import { Divider } from '@strapi/parts/Divider';
-import { useIntl } from 'react-intl';
-import { Tabs, Tab, TabGroup, TabPanels, TabPanel } from '@strapi/parts/Tabs';
-import { FromUrlForm } from './FromUrlForm';
-import { FromComputerForm } from './FromComputerForm';
-import { getTrad } from '../../utils';
+import { ModalLayout } from '@strapi/parts/ModalLayout';
+import { AddAssetStep } from './AddAssetStep/AddAssetStep';
+import { PendingAssetStep } from './PendingAssetStep/PendingAssetStep';
+
+const Steps = {
```

Comment:
I think at some it would be easier for you to handle the navigation using the same idea of the stepper but with an object,

```
const Steps = {
 AddAsset: {
  prev: null // closes the modal,
  next: 'PendingAsset',
},
PendingAsset: {
 prev: 'AddAssets',
 next: null,
}
}
```

It will make your entire navigation way easier from the beginning, then you could simply have some helpers like this one:

```
goNext = () => {
  if (step.next) {
     setState(Steps[next])
  } else {
   // close the modal
}
}
```